CORE-6690 consistent overrides

JasonSinclair95 opened the pull request:
This ticket is incomplete however I recommend we merge what has been done as a-lot has changed. This PR is to make changes so that the Corda workers can query an existing secret for both the db password and username
 
To test the changes the following should be put in your overrides 
```yaml
db:
  cluster:
    host: prereqs-postgresql 
    password:
      valueFrom:
        secretkeyRef:
          name: test-secret
          passwordKey: password 

    user:
      valueFrom: 
        secretkeyRef:
          name: test-secret2
          usernameKey: namename
```
For this deployment i created a secret for the username and one for the password. It was also tested using the same secret and no secret at all.

Commit:
CORE-6690 Add existing secret,secretKeyRef and key  option for db username and password

Commit:
CORE-6690 Add existing secret,secretKeyRef and key  option for db username and password

oscardalmau-r3 reviewed (dismissed):
LGTM

JasonSinclair95 committed:
Update values.yaml

davidcurrie commented on charts/corda/values.yaml:
```diff
@@ -60,13 +60,30 @@ db:
+          # -- the username secret name 
+          name: ""
+          # the username secret key 
+          usernameKey: ""
```
```suggestion
    user:
      # -- the cluster database user
      value: user
      # the cluster database user secret configuration; used in preference to value if name is set
      valueFrom: 
        # the cluster database user secret key reference
        secretKeyRef:
          # -- the cluster database user secret name 
          name: ""
          # -- the cluster database user secret key 
          key: ""
```

davidcurrie commented on charts/corda/values.yaml:
```diff
@@ -60,13 +60,30 @@ db:
+          name: ""
+          # the username secret key 
+          usernameKey: ""
     # -- the cluster database password (ignored if existingSecret is set, otherwise required)
```
```suggestion
    # the cluster database password configuration
```

davidcurrie commented on charts/corda/values.yaml:
```diff
@@ -60,13 +60,30 @@ db:
     # -- the cluster database port
     port: 5432
     # -- the cluster database user
```
```suggestion
    # the cluster database user configuration
```

davidcurrie commented on charts/corda/values.yaml:
```diff
@@ -60,13 +60,30 @@ db:
+          # -- the password secret name
+          name: ""
+          # -- the password secret key
+          passwordKey: ""
```
```suggestion
    password: 
      # -- the cluster database password
      value: "" 
      # the cluster database password secret configuration; used in preference to value if name is set
      valueFrom:
        # the cluster database password secret key reference
        secretKeyRef:
          # -- the cluster database password secret name
          name: ""
          # -- the cluster database password secret key
          key: ""
```

davidcurrie commented on charts/corda/values.schema.json:
```diff
@@ -246,13 +246,71 @@
-                            "title": "the cluster database user",
-                            "examples": [
-                                "user"
+                            "type": "object",
```
schema will need updating to reflect changes proposed in values

davidcurrie commented on charts/corda/templates/cluster-db-secret.yaml:
```diff
@@ -1,4 +1,4 @@
-{{- if not .Values.db.cluster.existingSecret }}
+{{- if not .Values.db.cluster.password.valueFrom.secretkeyRef.name }}
 apiVersion: v1
 kind: Secret
```
We should also put the user in this secret (which means the secret needs to be created if either the user or password is not specified via a user-provided secret.

JasonSinclair95 replied:
Okay

davidcurrie commented on charts/corda/templates/_helpers.tpl:
```diff
@@ -370,8 +379,13 @@ Cluster DB name
-Cluster DB secret name
+Cluster DB password secret 
 */}}
 {{- define "corda.clusterDbSecretName" -}}
```
```suggestion
{{- define "corda.clusterDbPasswordEnv" -}}
```

davidcurrie commented on charts/corda/templates/_helpers.tpl:
```diff
@@ -356,10 +356,19 @@ Cluster DB port
-Cluster DB user
+Cluster DB user secret 
 */}}
 {{- define "corda.clusterDbUser" -}}
```
```suggestion
{{- define "corda.clusterDbUseEnvr" -}}
```

davidcurrie commented on charts/corda/templates/_helpers.tpl:
```diff
@@ -356,10 +356,19 @@ Cluster DB port
+      name: {{ .Values.db.cluster.user.valueFrom.secretkeyRef.name }}
+      key: {{ .Values.db.cluster.user.valueFrom.secretkeyRef.usernameKey }}
+{{- else }}
+- name: PGUSER
+  value: {{ .Values.db.cluster.user.value }}
+{{- end }}
```
Will need to retrieve from secret if not set (as for password)

JasonSinclair95 replied:
That's what the function is. it looks for the value for the secret if set and if not it gets the default value which is user. Or is this not what you're suggesting @davidcurrie

davidcurrie replied:
Hopefully it made sense by the time you got to the comment about adding the user to the secret.

davidcurrie commented on charts/corda/templates/_helpers.tpl:
```diff
@@ -370,8 +379,13 @@ Cluster DB name
 
 {{/*
-Cluster DB secret name
+Cluster DB password secret 
```
```suggestion
Cluster DB password environment variable 
```

davidcurrie commented on charts/corda/templates/_helpers.tpl:
```diff
@@ -356,10 +356,19 @@ Cluster DB port
 
 {{/*
-Cluster DB user
+Cluster DB user secret 
```
```suggestion
Cluster DB user environment variable 
```

davidcurrie commented on charts/corda/templates/cluster-db-secret.yaml:
```diff
@@ -10,5 +10,5 @@ metadata:
-  password: {{ required "Must specify db.cluster.existingSecret or db.cluster.password" .Values.db.cluster.password | b64enc | quote }}
-{{- end }}
\ No newline at end of file
+  password: {{ required "Must specify db.cluster.password.valueFrom.secretRef.name or db.cluster.password.value" .Values.db.cluster.password.value | b64enc | quote }}
```
the `password` key here needs to use the `key` name specified in the values (or default to `password`)

JasonSinclair95 replied:
The default of this value .Values.db.cluster.password.value  is password @davidcurrie

davidcurrie replied:
I'm talking about the key here though i.e. the bit before the colon at the start of the line. You're always calling it `password` here but when you retrieve it, you use the value from `secretRef.key`.

JasonSinclair95 replied:
Could we have a conversation regarding this if statement please

JasonSinclair95 replied:
The key for the password should always be secret here, as this secret only gets created if SecretRef.name is not set. I don't see how any of the values are retrieved from SecretRef for this secret. @davidcurrie

JasonSinclair95 committed:
Update charts/corda/templates/_helpers.tpl

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

JasonSinclair95 committed:
Update charts/corda/templates/_helpers.tpl

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

JasonSinclair95 committed:
Update charts/corda/values.yaml

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

JasonSinclair95 committed:
Update charts/corda/values.yaml

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

JasonSinclair95 committed:
Update charts/corda/values.yaml

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

JasonSinclair95 committed:
Update charts/corda/values.yaml

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

JasonSinclair95 committed:
Update charts/corda/values.yaml

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

JasonSinclair95 committed:
Update charts/corda/templates/_helpers.tpl

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

JasonSinclair95 committed:
Update charts/corda/templates/_helpers.tpl

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

Commit:
CORE-6690 Add existing secret,secretKeyRef and key  option for db username and password

Commit:
CORE-6690 Add existing secret,secretKeyRef and key  option for db username,password and intialAdmin username,password

Commit:
CORE-6690 Add existing secret,secretKeyRef and key  option for db username,password and intialAdmin username,password

Commit:
cretaed init container to build sasl jaas.conf rather that get data from secret

Commit:
updated sasl password and username configuration

Commit:
updated sasl truststore secret and password configuration

Commit:
updated sasl truststore secret and password configuration

Commit:
CORE-6690 update json schema

Commit:
CORE-6690 update json schema

Commit:
CORE-6690 Review updates

yarivus commented on charts/corda/templates/_helpers.tpl:
```diff
@@ -221,6 +221,7 @@ Worker environment variables
 - name: CONSOLE_LOG_FORMAT
   value: {{ .Values.logging.format | quote }}
 - name: CONSOLE_LOG_LEVEL
+<<<<<<< HEAD
```
bad merge

yarivus reviewed (dismissed):
LGTM

Commit:
CORE-6690 Move clientImage to bootstrap

The DB client is only used as part of bootstrap so the configuration
should be defined there.

Commit:
CORE-6690 Re-add workerEnv to crypto worker

Commit:
CORE-6690 Additional review updates

oscardalmau-r3 reviewed (approved):
LGTM

ronanbrowne reviewed (approved):
LGTM
